core/domain/question_services.py

```diff
@@ -137,7 +137,10 @@ def get_questions_and_skill_descriptions_by_skill_ids(
     question_skill_links = [
         get_question_skill_link_from_model(model, skill_descriptions[index])
         for index, model in enumerate(question_skill_link_models)]
-    question_ids = [obj.question_id for obj in question_skill_links]
+    question_ids = []
```

Comment:
An easier way to do this is to use a set no?

Reply:
It's important to keep the order of question_ids because the index of question_ids should match skill_descriptions (for frontend use in the questions list), but the order of a set is random.

Reply:
But if that's the case why do you have the "if" statement below? If the question_ids list length is not the same as the question_skill_links list length then you can't match them up anyway.

Reply:
Actually this function will be replaced by another function in my milestone1.1 PR. This function is used by question players, so skill_descriptions is never used and I decided to get rid of it. But what I said above apply to `get_question_summaries_and_skill_descriptions` function below. For that one, the length of question_ids and skill_descriptions are matched.

Reply:
So this PR should only go in after your milestone 1.1 PR? We should make a note of that in the description... what PRs this is blocking on. (It can't go in as-is, at least this part, because it doesn't match the docstring...)